tests/cli/test_main_notices.py

```diff
@@ -277,11 +277,16 @@ def test_notices_appear_once_when_running_decorated_commands(
         "conda.notices.fetch.get_notice_responses", wraps=fetch.get_notice_responses
     )
 
+    if context.solver == "classic":
```

Comment:
@jaimergp the package missing message is different between classic and libmamba. Does this approach look good?

Reply:
I was about to comment that the message difference is explained by the exception class logic. I think I remember that it will be different if the exception is passed the channels as arguments or not. If no channels are passed, the message is about the target environment (e.g. when you conda remove or conda update a package that is not installed).

So check where we are raising PackagesNotFoundError with channels  when we should not.

Reply:
I added a TODO related to this comment. I'll continue to work on tests just to see which ones I could work through.

Reply:
Ok I took another look at this because it rang a bell. I've seen this before and documented it at https://github.com/conda/conda/issues/12197. For now, I'd say it's ok to have this conditional message. In the future, we should address the mixed exceptions issue and fix conda classic's behaviour. IOW, libmamba is right here and classic is wrong, not the other way around.

Reply:
Resolved and added link to issue in test